src/FakeClient/GraphicsCaptureEventProcessor.h

```diff
@@ -0,0 +1,230 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FAKE_CLIENT_GRAPHICS_CAPTURE_EVENT_PROCESSOR_H_
+#define FAKE_CLIENT_GRAPHICS_CAPTURE_EVENT_PROCESSOR_H_
+
+#include "CaptureClient/CaptureEventProcessor.h"
+#include "Flags.h"
+#include "OrbitBase/WriteStringToFile.h"
+
+namespace {
+template <typename Distribution, typename UnaryOperation>
+void ForEachCentile(uint32_t numCentiles, const Distribution& distribution,
+                    UnaryOperation&& operation) {
+  uint32_t populationSize = std::accumulate(std::begin(distribution), std::end(distribution), 0);
+  size_t currentBucket = 0;
+  uint32_t runningCount = 0;
+  for (uint32_t centile = 1; centile <= numCentiles; ++centile) {
+    while (currentBucket < distribution.size() &&
+           // checks if runningCount is less than centile/numCentiles of the
+           // population size, but keeps the calculation within the integral
+           // domain by scaling with numCentiles
+           (runningCount * numCentiles) < (populationSize * centile)) {
+      runningCount += distribution[currentBucket];
+      ++currentBucket;
+    }
+    operation(currentBucket);
+  }
+}
+}  // namespace
+
+namespace orbit_fake_client {
+
+// This implementation of CaptureEventProcessor mostly discard all events it receives, except for
+// keeping track of the calls to the frame boundary function and GPU queue submissions.
+class GraphicsCaptureEventProcessor : public orbit_capture_client::CaptureEventProcessor {
+ public:
+  void ProcessEvent(const orbit_grpc_protos::ClientCaptureEvent& event) override {
+    switch (event.event_case()) {
+      case orbit_grpc_protos::ClientCaptureEvent::kFunctionCall:
+        ProcessFunctionCall(event.function_call());
+        break;
+      case orbit_grpc_protos::ClientCaptureEvent::kGpuQueueSubmission:
+        ProcessGPUQueueSubmission(event.gpu_queue_submission());
+        break;
+      default:
+        break;
+    }
+  }
+
+  void ProcessGPUQueueSubmission(const orbit_grpc_protos::GpuQueueSubmission& submission) {
```

Comment:
ProcessGPUQueueSubmission: GPU -> Gpu
CalculateGPUTimes: GPU -> Gpu
CalculateCPUTimes: CPU -> Cpu
WriteToCSVFile: CSV -> Csv
kCPUFrameTimeFilename: CPU -> Cpu
kGPUFrameTimeFilename: GPU -> Gpu
kCSVHeader: CSV -> Csv

Styleguide: "prefer to capitalize abbreviations as single words, e.g., StartRpc() rather than StartRPC()"
Yeah, exactly the opposite of golang...

Reply:
Done